extension/configintegrationextension/config.go

```diff
@@ -0,0 +1,47 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package configintegrationextension // import "github.com/elastic/opentelemetry-collector-components/extension/configintegrationextension"
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap/xconfmap"
+
+	"github.com/elastic/opentelemetry-collector-components/internal/integrations"
```

Comment:
Depending on an internal package of a different module is usually a rather bad idea (see https://github.com/open-telemetry/opentelemetry-go/issues/3846).

Reply:
Happy to change the approach to share this code.
Would it make sense to move it to a public package under this repository?
Something like `github.com/elastic/opentelemetry-collector-components/pkg/integrations`?

I guess another alternative also mentioned in the linked issue would be to copy the code with gogenerate, but having it in a different [public] package would sound better to me.

Reply:
I think a different public package seems better.

Reply:
Ok, moving it in https://github.com/elastic/opentelemetry-collector-components/pull/484.

Reply:
Updated.